lib/ddtrace/contrib/rails/core_extensions.rb

```diff
@@ -9,15 +9,33 @@ def patch_renderer
     end
 
     def patch_renderer_render_template
-      if defined?(::ActionView::Renderer)
-        ::ActionView::Renderer.class_eval do
+      if defined?(::ActionView::TemplateRenderer)
+        ::ActionView::TemplateRenderer.class_eval do
+          alias_method :render_without_datadog, :render
           alias_method :render_template_without_datadog, :render_template
-          def render_template(*args, &block)
-            ActiveSupport::Notifications.instrument('start_render_template.action_view')
-            render_template_without_datadog(*args, &block)
+
+          def render(context, options)
+            # create a tracing context and start the rendering span
+            @tracing_context = {}
+            ::ActiveSupport::Notifications.instrument('start_render_template.action_view', tracing_context: @tracing_context)
+            render_without_datadog(context, options)
+          rescue StandardError => e
+            # attach the exception to the tracing context if any
+            @tracing_context[:exception] = e
+            raise e
+          ensure
+            # ensure that the template `Span` is finished even during exceptions
+            ::ActiveSupport::Notifications.instrument('finish_render_template.action_view', tracing_context: @tracing_context)
```

Comment:
Whatever happens we close the span, even if in some cases (i.e. critical exception that doesn't trigger a rendering) we may not have all tags set. This is the best safe-guard we can have for the worst case: not detailed but at least properly finished spans (with the exception attached).